### PR TITLE
Replace __threadlocal with thread_local.

### PR DESCRIPTION
**Pull request by KousukeItsagame**
Requires GCC 4.8.x See cxx11.

### CODE REVIEW

**Comment by sudonim1** [comment]
From what I read this absolutely murders performance on every reference to a thread local variable.  Unsure how it affects MSVC.

**Comment by KousukeItsagame** [comment]
It compiles on visual studio 2015.
I do not know how to check te performance of it in windows
but I added this as it seems to do the same but is newer.

**Comment by KousukeItsagame** [comment]
I accidently mad another pull request.
Can you delete it?

**Comment by uyjulian** [comment]
Just close it.

**Comment by refractionpcsx2** [comment]
Closed the other one.

**Comment by KousukeItsagame** [comment]
thanks 👍

**Comment by sudonim1** [comment]
So we've gone from doesn't compile in msvc 2013 to doesn't compile anywhere?

**Comment by refractionpcsx2** [comment]
..\..\src\Utilities\Console.cpp(26): error C2144: syntax error: 'int' should be preceded by ';' [C:\projects\pcsx2\common\build\Utilities\utilities.vcxproj]
         ..\..\src\Utilities\Console.cpp(26): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\projects\pcsx2\common\build\Utilities\utilities.vcxproj]
         ..\..\src\Utilities\Console.cpp(29): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\projects\pcsx2\common\build\Utilities\utilities.vcxproj]
         ..\..\src\Utilities\Console.cpp(29): error C2086: 'int __threadlocal': redefinition [C:\projects\pcsx2\common\build\Utilities\utilities.vcxproj]
         ..\..\src\Utilities\Console.cpp(29): error C2146: syntax error: missing ';' before identifier 'ConsoleColors' [C:\projects\pcsx2\common\build\Utilities\utilities.vcxproj]
         ..\..\src\Utilities\Exceptions.cpp(53): error C2144: syntax error: 'int' should be preceded by ';' [C:\projects\pcsx2\common\build\Utilities\utilities.vcxproj]
         ..\..\src\Utilities\Exceptions.cpp(53): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\projects\pcsx2\common\build\Utilities\utilities.vcxproj]

**Comment by refractionpcsx2** [comment]
In future please compile your code before submitting it please :)

**Comment by sudonim1** [comment]
What exactly is the point of using the C++11 specifier if we have to hide it behind a lot of macros and precompiler conditions anyway?

**Comment by KousukeItsagame** [comment]
Maybe not now, But in the future?
the macro's are only for visual studio 2013
it's the only build target that doesn't work with local_thread

I'm finishing the work anyway :)
I'm aware it might not be used though

@refractionpcsx2 ok

File: common/include/Utilities/TlsVariable.inl [hunk 1]
```diff
@@ -18,7 +18,11 @@
 #include "Threading.h"
 
 #if PCSX2_THREAD_LOCAL
-#	define DeclareTls(x) __threadlocal x
+#	if _MSC_VER = 1800
```
Comment (willkuer) [inline common/include/Utilities/TlsVariable.inl]: = or ==?

**Comment by gregory38** [comment]
```
Maybe not now, But in the future?
```
I don't think compiler will stop supporting their builtin. Is just a different keyword to parse. In the enterprise world, people just began to use C++11.

On your PR, if I understand correctly, `__threadlocal` is local PCSX2 macro. So you only need to update `Pcsx2Defs.h`.

Honestly, I think it will best to wait VS2017 release (or whatever it is called). Then we could drop VS2013 and uses fully the new syntax sugar of C++11.

**Comment by gregory38** [comment]
Sorry, we are in the middle of automatic reformating the code with clang-format (the common subdirectory). You will need to rebase, and you will need to format the code with clang-format.

**Comment by KousukeItsagame** [comment]
I'll agree to postpone it.
I'll start again when it's more feasable in the future :)

**Comment by gregory38** [comment]
For the record, I plan to use clang-tidy to upgrade the db to C++11 syntax sugar. However it will be done after the complete reformating. And likely after the next release.

**Pull request closed without merging**